src/cache.rs

```diff
@@ -0,0 +1,283 @@
+//! Cache
```

Comment:
I currently don't know why caching should belong directly to IPLD, it sounds like an application level kind of thing. Though atm I also don't have strong objections against having it here.